posthog/clickhouse/migrations/0045_add_insertion_timestamp_to_events.py

```diff
@@ -0,0 +1,30 @@
+from infi.clickhouse_orm import migrations
+
+from posthog.clickhouse.client.migration_tools import run_sql_with_exceptions
+from posthog.client import sync_execute
+from posthog.models.event.sql import (
+    EVENTS_TABLE_JSON_MV_SQL,
+    KAFKA_EVENTS_TABLE_JSON_SQL,
+)
+from posthog.settings import CLICKHOUSE_CLUSTER
+
+ADD_COLUMNS_BASE_SQL = """
+ALTER TABLE {table}
+ON CLUSTER '{cluster}'
+ADD COLUMN IF NOT EXISTS inserted_at DateTime64(6, 'UTC') DEFAULT NOW()
+"""
+
+
+def add_columns_to_required_tables(_):
+    sync_execute(ADD_COLUMNS_BASE_SQL.format(table="events", cluster=CLICKHOUSE_CLUSTER))
+    sync_execute(ADD_COLUMNS_BASE_SQL.format(table="writable_events", cluster=CLICKHOUSE_CLUSTER))
+    sync_execute(ADD_COLUMNS_BASE_SQL.format(table="sharded_events", cluster=CLICKHOUSE_CLUSTER))
+
+
+operations = [
+    run_sql_with_exceptions(f"DROP TABLE IF EXISTS events_json_mv ON CLUSTER '{CLICKHOUSE_CLUSTER}'"),
```

Comment:
I feel like we shouldn't need to drop and recreate here, did you have issues if you didn't do this.

Reply:
Locally, all 3 ALTER TABLE queries work without dropping. I copied this from a previous migration (0030).

Reply:
I vote we remove the drop and recreate, just to keep the number of moving parts down, (and I don't see what the value is, only potential downsides)

Reply:
Yeah, I'm fine with that 👍

Reply:
I've dropped the drops.